sirepo/template/radia.py

```diff
@@ -418,15 +449,45 @@ def _read_or_generate(geom_id, data):
         return get_application_data(data)
 
 
-def _save_fm_sdds(name, f_data, file_path):
+def _read_solution(sim_id):
+    try:
+        return _read_path(sim_id, 'solution')
+    except KeyError:
```

Comment:
_read_path should tell you the result. KeyError could be raised for other reasons (ie. other defects) and therefore it's too coarse. Better to return an explicit "no solution" (None) result and test for that here.